haskell/private/haddock_wrapper.sh

```diff
@@ -36,4 +36,10 @@ do
     fi
 done
 
-haddock "${extra_args[@]}" "$@"
+# Override TMPDIR to prevent race conditions on certain platforms.
+# BSD and GNU mktemp are very different; attempt GNU first
+TEMP=$(mktemp -d 2>/dev/null || mktemp -d -t 'haddock_wrapper')
+trap cleanup 1 2 3 6
+cleanup() { rmdir "$TEMP"; }
+TMPDIR=$TEMP haddock "${extra_args[@]}" "$@"
```

Comment:
There should be a link to an upstream haddock issue here.

Reply:
Sorry, this should have been a comment, not a blocker. Thanks to @judah for catching it and merging!